synapse/util/threepids.py

```diff
@@ -28,9 +31,20 @@ def check_3pid_allowed(hs, medium, address):
         address (str): address within that medium (e.g. "wotan@matrix.org")
             msisdns need to first have been canonicalised
     Returns:
-        bool: whether the 3PID medium/address is allowed to be added to this HS
+        defered bool: whether the 3PID medium/address is allowed to be added to this HS
     """
 
+    if hs.config.check_is_for_allowed_local_3pids:
+        data = yield hs.get_simple_http_client().get_json(
+            "https://%s%s" % (
+                hs.config.check_is_for_allowed_local_3pids,
+                "/_matrix/identity/api/v1/discover_urls"
+            ),
+            {'medium': medium, 'address': address}
+        )
+        defer.returnValue(data['hs_url'] + "/" == hs.config.public_baseurl)
```

Comment:
This is going to be prone to non-obvious breakage depending on if slashes are added or not. Can we not use the server names here?